app/src/main/java/com/example/soswedding/ui/SignIn/SignInFragment.java

```diff
@@ -66,25 +79,51 @@ public void onClick(View view) {
         });
     }
     private void setupSignUpOnClickEvent(){
-
+        signUpBtn.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view){
+                onSignUpClicked(view);
+            }
+        });
     }
     private void setupSignInOnClickEvent(){
-
+        loginBtn.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view){
+                onLoginClick(view);
+            }
+        });
     }
 
+    public void onLoginClick(View view){
+        if(emailInput.length()>0 && passwordInput.length()>0){
+            verifyLogIn(emailInput.getText().toString(),passwordInput.getText().toString() );
+        } else {
+            popUp("Please Fill up the username and password fields before.");
+        }
+    }
 
+    private void verifyLogIn(String email, String password){
+        auth.signInWithEmailAndPassword(email, password).addOnCompleteListener(getActivity() , new OnCompleteListener<AuthResult>() {
+            @Override
+            public void onComplete(@NonNull Task<AuthResult> task) {
```

Comment:
A method inside a method? :0

Reply:
In this case it is an event handler method, oncomplete and is required when using the signInWithEmailAndPassword(). However, I do agree,  it could be better to extract in the class and then add it as an argument, what do you think

Reply:
yup :3